reflect.go

```diff
@@ -890,6 +891,37 @@ func (r *Reflector) typeName(t reflect.Type) string {
 	return t.Name()
 }
 
+// Split on commas that are not preceded by `\`.
+// This way, we prevent splitting regexes
+func splitOnUnescapedCommas(tagString string) []string {
+	splitOn := regexp.MustCompile(`[^\\](,)`)
+	commaIndices := splitOn.FindAllStringIndex(tagString, -1)
```

Comment:
Could we not use `regexp.Split` here? (https://pkg.go.dev/regexp#Regexp.Split)

Reply:
Tried it, but the regex `[^\\](,)` matches not only the comma, but also the letter immediately preceding it. If we'd use regex.Split in this situation, the tag string `"ayy=lmao,hello=world"` would be split into `["ayy=lma", "hello=world"]`. 

We could in theory do a regex that _only_ matches the comma, but that would require a regex engine supporting lookbehind. The default go regex engine (re2) does not do this. 
We could go ahead and use one that supports this (e.g. https://github.com/dlclark/regexp2), but doing so would add an entire dependency for one single function call. That's why I decided to implement splitting myself.

Reply:
Of course 🤦  I'm just a bit worried its a beast of a function for something you'd expect to be straight forward. One final idea: You could split as usual (`strings.Split(tagString, ",")`) and then iterate through the results and detect any item that ends in `\`. If it does, you can just concatenate the next entry. Might be a little less complex!

Reply:
Okay, changed it! Seems less complex to me now!